### PR TITLE
Support show create tag/edge xxx, show create space xxx SQL

### LINKED ISSUE #439: Support show create tag/edge xxx,   show create space xxx SQL
Current， desc tag/edge,  not  display table-level attributes, only display column attributes.
so, use show create tag/edge SQL, to display column attributes and table-level attributes.
e.g. comment，ttl_duration,  ttl_col, character set, engines.

### PR DESCRIPTION
**Pull request by steppenwolfyuetong**
Support show create tag/edge xxx,  show create space xxx SQL
Close #439

This PR contain TTL_schema (PR #422). Because it depends on the attributes of the schema.

### CODE REVIEW

**Comment by steppenwolfyuetong** [comment]
Please code review, thx.

File: src/graph/CreateTagExecutor.cpp [hunk 1]
```diff
@@ -18,25 +18,97 @@ CreateTagExecutor::CreateTagExecutor(Sentence *sentence,
...
         column.name = *spec->name();
         column.type.type = columnTypeToSupportedType(spec->type());
-        schema.columns.emplace_back(std::move(column));
+        schema_.columns.emplace_back(std::move(column));
     }
 
-    auto future = mc->createTagSchema(spaceId, *name, schema);
+    // set schema_.schema_prop default
+    // 0 means infinity
+    schema_.schema_prop.set_ttl_duration(0);
```
Comment (laura-ding) [inline src/graph/CreateTagExecutor.cpp]: It can init in the constructor of schema_prop

File: src/graph/DescribeSpaceExecutor.cpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
-/* Copyright (c) 2019 - present, VE Software Inc. All rights reserved
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
  *
- * This source code is licensed under Apache 2.0 License
- *  (found in the LICENSE.Apache file in the root directory)
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
```
Comment (laura-ding) [inline src/graph/DescribeSpaceExecutor.cpp]: well done

File: src/graph/CreateEdgeExecutor.cpp [hunk 3]
```diff
@@ -18,25 +18,96 @@ CreateEdgeExecutor::CreateEdgeExecutor(Sentence *sentence,
...
+                    break;
+                case SchemaPropItem::TTL_COL:
+                    status = setTTLCol(schemaProp);
+                    if (!status.ok()) {
+                        return status;
+                    }
+                    break;
+            }
+        }
+        if (schema_.schema_prop.ttl_duration != 0) {
```
[the remark this answers is not in the document]
Comment (steppenwolfyuetong) [reply src/graph/CreateEdgeExecutor.cpp]: no, no. when onlyse set ttl_col,  the ttl_duration is default 0. 

This situation is implicit ttl, which is forbidden here.  

SQL like：
`create tag t ( a int) ttl_col = a;`  we don't support.

File: src/graph/CreateTagExecutor.cpp (hunk 1 above)
Comment (steppenwolfyuetong) [reply src/graph/CreateTagExecutor.cpp]: 👍👍👍，good  catch！

File: src/graph/CreateTagExecutor.cpp [hunk 4]
```diff
@@ -18,25 +18,97 @@ CreateTagExecutor::CreateTagExecutor(Sentence *sentence,
...
+                case SchemaPropItem::TTL_COL:
+                    status = setTTLCol(schemaProp);
+                    if (!status.ok()) {
+                        return status;
+                    }
+                    break;
+            }
+        }
+
+        if (schema_.schema_prop.ttl_duration != 0) {
```
[the remark this answers is not in the document]
Comment (steppenwolfyuetong) [reply src/graph/CreateTagExecutor.cpp]: when onlyse set ttl_col, the ttl_duration is default 0.

This situation is implicit ttl, which is forbidden here.

SQL like：
create tag t ( a int) ttl_col = a; we don't support.

**Comment by steppenwolfyuetong** [comment]
rebase master. please code review again, thx

File: src/parser/AdminSentences.cpp [hunk 5]
```diff
@@ -96,12 +102,14 @@ std::string CreateSpaceSentence::toString() const {
     return buf;
 }
 
+
 std::string DropSpaceSentence::toString() const {
```
Comment (whitewum) [inline src/parser/AdminSentences.cpp]: const std::string

**Comment by steppenwolfyuetong** [comment]
> is this a mixed pr? should I wait for #422?

Yeah, the pr contains PR #422.
After pr 422 merged, I will update this PR.

File: src/parser/AdminSentences.cpp (hunk 5 above)
Comment (steppenwolfyuetong) [reply src/parser/AdminSentences.cpp]: Never qualify with const when you return by value.

**Comment by steppenwolfyuetong** [comment]
Jenkins, go

**Comment by steppenwolfyuetong** [comment]
Jenkins, go

**Comment by steppenwolfyuetong** [comment]
Please review，thx

**Comment by steppenwolfyuetong** [comment]
Jenkins, go

File: docs/nGQL-tutorial.md [hunk 6]
```diff
@@ -98,6 +98,15 @@ SHOW SPACES
 ----------------
 ```
 
+```
+SHOW CREATE SPACES myspace_test
+------------------------------------------------------------------------
+| Space        | Create Space                                           |
+------------------------------------------------------------------------
+| myspace_test | CREATE SPACE s (partition_num = x, replica_factor = x) |
```
Comment (whitewum) [inline docs/nGQL-tutorial.md]: CREATE SPACE `myspace_test` (partition_num = x, replica_factor = y)

BTW, this looks a bit different from mysql manual.

Comment (steppenwolfyuetong) [reply docs/nGQL-tutorial.md]: Yeah, I update it  following the output of nebula

Comment (steppenwolfyuetong) [reply docs/nGQL-tutorial.md]: Good catch! 👍

**Comment by steppenwolfyuetong** [comment]
Jenkins, go!

**Comment by steppenwolfyuetong** [comment]
fix and rebase master.

**Comment by steppenwolfyuetong** [comment]
Jenkins, go!

File: src/graph/ShowExecutor.cpp [hunk 7]
```diff
@@ -226,6 +233,214 @@ void ShowExecutor::showEdges() {
...
+        resp_->set_rows(std::move(rows));
+
+        DCHECK(onFinish_);
+        onFinish_();
+    };
+
+    auto error = [this] (auto &&e) {
+        LOG(ERROR) << "Exception caught: " << e.what();
+        DCHECK(onError_);
+        onError_(Status::Error(folly::stringPrintf("Internal error : %s",
```
Comment (laura-ding) [inline src/graph/ShowExecutor.cpp]: The specific reason for internal error doesn't need to be known by the user

File: src/graph/ShowExecutor.cpp [hunk 8]
```diff
@@ -226,6 +233,214 @@ void ShowExecutor::showEdges() {
...
+            buf += ",\n";
+        }
+
+        if (!schema.columns.empty()) {
+            buf.resize(buf.size() -2);
+            buf += "\n";
+        }
+        buf += ") ";
+        nebula::cpp2::SchemaProp prop = schema.schema_prop;
+        buf += "ttl_duration = ";
```
Comment (laura-ding) [inline src/graph/ShowExecutor.cpp]: When the user does not have Settings, there should be no need to display them

File: src/graph/test/SchemaTest.cpp [hunk 9]
```diff
@@ -197,6 +212,54 @@ TEST_F(SchemaTest, metaCommunication) {
         };
         ASSERT_TRUE(verifyResult(resp, expected));
     }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string query = "SHOW CREATE TAG person";
```
Comment (laura-ding) [inline src/graph/test/SchemaTest.cpp]: Execute the resulting command again to see if it works

File: src/graph/ShowExecutor.cpp (hunk 7 above)
Comment (steppenwolfyuetong) [reply src/graph/ShowExecutor.cpp]: An exception has occurred here, we should know what is exception and be easy to handle it.

File: src/graph/ShowExecutor.cpp [hunk 10]
```diff
@@ -226,6 +233,214 @@ void ShowExecutor::showEdges() {
...
+        resp_->set_rows(std::move(rows));
+
+        DCHECK(onFinish_);
+        onFinish_();
+    };
+
+    auto error = [this] (auto &&e) {
+        LOG(ERROR) << "Exception caught: " << e.what();
+        DCHECK(onError_);
+        onError_(Status::Error(folly::stringPrintf("Internal error : %s",
```
[the remark this answers is not in the document]
Comment (steppenwolfyuetong) [reply src/graph/ShowExecutor.cpp]: ditto

File: src/graph/test/SchemaTest.cpp (hunk 9 above)
Comment (steppenwolfyuetong) [reply src/graph/test/SchemaTest.cpp]: Here mainly shows the properties of schema.

**Comment by steppenwolfyuetong** [comment]
Fix and rebase master. Please review again, thx.

**Comment by steppenwolfyuetong** [comment]
Jenkins, go

**Comment by steppenwolfyuetong** [comment]
Jenkins, go

**Comment by Amber1990Zhang** [comment]
test

**Comment by steppenwolfyuetong** [comment]
Rebase master, please review, thx.

File: src/graph/ShowExecutor.cpp [hunk 11]
```diff
@@ -19,7 +19,14 @@ ShowExecutor::ShowExecutor(Sentence *sentence,
 
 
 Status ShowExecutor::prepare() {
-    return Status::OK();
+    if (sentence_->showType() == ShowSentence::ShowType::kShowTags ||
```
Comment (laura-ding) [inline src/graph/ShowExecutor.cpp]: please see pr #740

Comment (steppenwolfyuetong) [reply src/graph/ShowExecutor.cpp]: Good catch, 👍👍

**Comment by steppenwolfyuetong** [comment]
Jenkins, go!

**Comment by steppenwolfyuetong** [comment]
Fix and rebase master. Please review again, thx.

**Pull request merged**